File: articles/history-of-algorithms/prolog/euclidean_algorithm.pl
gcd_algorithm(A, 0, G) :- G is abs(A).
gcd_algorithm(A, B, G) :-
    B =\= 0,
    R is A mod B,
    gcd_algorithm(B, R, G).

% Query example:
% ?- gcd_algorithm(252, 105, G).

File: articles/history-of-algorithms/fortran/euclidean_algorithm.f90
program euclidean_algorithm_example
  implicit none
  integer :: a, b, r
  a = 252
  b = 105
  do while (b /= 0)
    r = mod(a, b)
    a = b
    b = r
  end do
  print *, "gcd=", abs(a)
end program euclidean_algorithm_example
